catroid/src/org/catrobat/catroid/physics/shapebuilder/PhysicsShapeBuilder.java

```diff
@@ -106,4 +109,27 @@ private String getKey(LookData lookData, float scaleFactor) {
 
 		return scaledShapes.toArray(new Shape[scaledShapes.size()]);
 	}
+
+	public static Vector2 scaleCoordinate(Vector2 vertex, float scaleFactor) {
+		if (Math.abs(scaleFactor) < 0.001f) {
+			vertex = vertex.mul(scaleFactor);
+		} else if (Math.abs(scaleFactor - 1.0f) > 0.001f) {
+			Vector2 offset = new Vector2();
+			offset.x = vertex.x < -0.001f ? -OFFSET_VALUE : vertex.x > 0.001f ? OFFSET_VALUE : 0.0f;
+			offset.y = vertex.y < -0.001f ? -OFFSET_VALUE : vertex.y > 0.001f ? OFFSET_VALUE : 0.0f;
+			vertex = vertex.add(offset).mul(scaleFactor).sub(offset);
+		}
+		return vertex;
+	}
+
+	public static float scaleCoordinate (float coord, float factor) {
+		if (Math.abs(factor) < 0.001f) {
+			coord = coord*factor;
+		} else if (Math.abs(factor - 1.0f) > 0.001f) {
+			float offset;
+			offset = coord < -0.001f ? -OFFSET_VALUE : coord > 0.001f ? OFFSET_VALUE : 0.0f;
```

Comment:
A method for that would be better. It's used at least 3 times and a good name would clarify what it does.